crates/ty_python_semantic/src/types.rs

```diff
@@ -1054,6 +1054,16 @@ impl<'db> Type<'db> {
             || self.is_literal_string()
     }
 
+    pub(crate) fn is_union_with_single_valued(&self, db: &'db dyn Db) -> bool {
+        self.into_union().is_some_and(|union| {
+            union
+                .elements(db)
+                .iter()
+                .any(|ty| ty.is_single_valued(db) || ty.is_bool(db) || ty.is_literal_string())
+        }) || self.is_bool(db)
+            || self.is_literal_string()
```

Comment:
We include `bool` and `LiteralString` here as types that make a type be a "union with single valued", but where we actually use this method, we only really handle unions, and we don't do any special handling for `bool` or `LiteralString`. So I don't think there's currently any benefit to including them here.

Maybe we could add some tests that have `bool` or `LiteralString` types on the LHS -- I think that would illuminate cases that we'd want to handle. They could stay TODOs in this PR, if they aren't easy to handle.

Reply:
https://github.com/astral-sh/ruff/pull/20164/files#diff-4d769c34388fdd9011945264bed8fb2199b60ffefcbaa09d0804e4e6c347fd4dR134
added some TODOs in the in.md. If the tests looks good to you, I will take a look tomorrow to see if it's doable for me in the near future.